src/Environment.php

```diff
@@ -379,8 +393,10 @@ public function loadTemplate(string $cls, string $name, ?int $index = null): Tem
             if (!class_exists($cls, false)) {
                 $source = $this->getLoader()->getSourceContext($name);
                 $content = $this->compileSource($source);
-                $this->cache->write($key, $content);
-                $this->cache->load($key);
+                if (!isset($this->hotCache[$name])) {
```

Comment:
Do not pollute the cache with a temporary class that will never be reused in another/new PHP process.